packages/wonder-blocks-popover/components/popover-content.js

```diff
@@ -122,29 +135,75 @@ export default class PopoverContent extends React.Component<Props> {
-                        </HeadingSmall>
-                        <Body>{content}</Body>
-                    </View>
-                </View>
-
-                {actions && <View style={styles.actions}>{actions}</View>}
-            </PopoverContentCore>
+            <PopoverContext.Consumer>
+                {({close, placement}) => {
+                    if (close && onClose) {
+                        throw new Error(
+                            "You've specified 'onClose' on the content when using Popover. Please specify 'onClose' on the Popover instead",
+                        );
+                    }
+
+                    // illustration popover can't be placed horizontally
+                    if (
+                        image &&
+                        (placement === "left" || placement === "right")
+                    ) {
+                        throw new Error(
+                            "'image' can only be vertically placed. You can fix this by either changing `placement` to `top` or `bottom` or removing the `image` prop inside `content`.",
+                        );
+                    }
+
+                    return (
+                        <PopoverContentCore
+                            color={emphasized ? "blue" : "white"}
+                            closeButtonLight={image && placement === "top"}
+                            closeButtonLabel={closeButtonLabel}
+                            closeButtonVisible={closeButtonVisible}
+                            onClose={onClose}
+                            style={style}
+                            testId={testId}
+                        >
+                            <View style={!!icon && styles.withIcon}>
+                                {image && (
+                                    <View
+                                        style={[
+                                            styles.image,
+                                            placement === "bottom" &&
+                                                styles.imageToBottom,
+                                        ]}
+                                    >
+                                        {image}
+                                    </View>
+                                )}
+
+                                {icon && (
+                                    <StyledImage
+                                        style={styles.icon}
+                                        src={icon}
+                                    />
+                                )}
+
+                                <View style={styles.text}>
+                                    <HeadingSmall style={styles.title}>
+                                        {title}
+                                    </HeadingSmall>
+                                    <Body>{content}</Body>
+                                </View>
+                            </View>
+
+                            {actions && (
```

Comment:
I often see this sort of inline optional rendering as a refactor smell and prefer the use of `maybe...` functions for rendering optional pieces like this. It tends to make for easier to read code. What do you think?

```
maybeRenderActions() {
    const {actions} = this.props;
    if (!actions) {
       return null;
    }
    return (<ComponentStuffToRender />);
}
```

then the main render just has:

```
...
{this.maybeRenderActions()}
...
```

Reply:
This goes for the icon and image stuff too.

Reply:
Sounds good to me :). Thanks for the suggestion!